### PR TITLE
gamehub: init at 0.16.0-1

### PR DESCRIPTION
**Pull request by pasqui23**
###### Motivation for this change
Gamehub is a game launcher supporting Steam,GOG and Humble Bundle

###### Things done

Compiles,but it cores dump with:
```
nix shell .#gamehub -c gamehub   
Gtk-Message: 16:03:18.564: Failed to load module "appmenu-gtk-module"
zsh: trace trap (core dumped)  nix shell .#gamehub -c gamehub
```

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pasqui23** [comment]
@LouisDK1 would https://github.com/NixOS/nixpkgs/pull/95412 help here?

**Comment by LouisDK1** [comment]
> @LouisDK1 would #95412 help here?

Yes it would. I'll try to make time to the fix up that pull request.

File: pkgs/games/gamehub/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "GameHub";
+  version = "0.16.0-1";
+  src = fetchFromGitHub {
+    owner = "tkashkin";
+    repo = pname;
+    rev = "${version}-master";
+    sha256 = "sha256-QcpqGydNkmAPrPxp/52+D6MS1I7k+CKHxr3/tpCukP8=";
+  };
+  nativeBuildInputs = [
+    meson ninja vala pkg-config desktop-file-utils
```
Comment (tcbravo) [inline pkgs/games/gamehub/default.nix:15]: The app starts for me when I add `wrapGAppsHook` here (even without `appmenu-gtk-module`).

Comment (pasqui23) [reply pkgs/games/gamehub/default.nix:15]: With `wrapGAppsHook` added to nativeBuildInputs I get the following error on run:
```
~/.nix-profile/bin/gamehub: /nix/store/9df65igwjmf2wbw0gbrrgair6piqjgmi-glibc-2.31/lib/libc.so.6: version `GLIBC_2.32' not found (required by /nix/store/fdbsckqrhfmwy73mck958pcw5z51wa3b-systemd-247.2/lib/libsystemd.so.0)
```

Comment (tcbravo) [reply pkgs/games/gamehub/default.nix:15]: Looks like it's looking for the wrong version of `glibc`. The version in the fork is [2.31](https://github.com/pasqui23/nixpkgs/blob/gamehub/pkgs/development/libraries/glibc/common.nix#L44), but unstable has been on [2.32](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/glibc/common.nix#L44) for a while, which would explain why it doesn't work (assuming you're on unstable, anyway). Could you try rebasing your branch on master?

Comment (pasqui23) [reply pkgs/games/gamehub/default.nix]: Yoou're right

File: pkgs/games/gamehub/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  propagatedBuildInputs = [
+    glib gtk3 libgee libsoup
+    json-glib sqlite webkitgtk libmanette libXtst
+  ];
+  meta = with stdenv.lib;{
+    description = "All your games in one place";
+    homepage = "https://tkashkin.tk/projects/gamehub/";
+    mantainers = with mantainers;[ pasqui23 ];
+    license = with licenses;[ gpl3Only ];
+    platforms = with platforms;[ linux ];
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:28]:
```suggestion
    mantainers = with mantainers; [ pasqui23 ];
    license = with licenses; [ gpl3Only ];
    platforms = with platforms; [ linux ];
```

File: pkgs/games/gamehub/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  nativeBuildInputs = [
+    meson ninja vala pkg-config
+    desktop-file-utils wrapGAppsHook
+
+  ];
+  propagatedBuildInputs = [
+    glib gtk3 libgee libsoup
+    json-glib sqlite webkitgtk libmanette libXtst
+  ];
+  meta = with stdenv.lib;{
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:23]:
```suggestion
  ];

  meta = with lib;{
```

File: pkgs/games/gamehub/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    repo = pname;
+    rev = "${version}-master";
+    sha256 = "sha256-QcpqGydNkmAPrPxp/52+D6MS1I7k+CKHxr3/tpCukP8=";
+  };
+  nativeBuildInputs = [
+    meson ninja vala pkg-config
+    desktop-file-utils wrapGAppsHook
+
+  ];
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:19]:
```suggestion
    desktop-file-utils wrapGAppsHook
  ];

  propagatedBuildInputs = [
```

File: pkgs/games/gamehub/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+stdenv.mkDerivation rec {
+  pname = "GameHub";
+  version = "0.16.0-1";
+  src = fetchFromGitHub {
+    owner = "tkashkin";
+    repo = pname;
+    rev = "${version}-master";
+    sha256 = "sha256-QcpqGydNkmAPrPxp/52+D6MS1I7k+CKHxr3/tpCukP8=";
+  };
+  nativeBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:14]:
```suggestion
  };

  nativeBuildInputs = [
```

File: pkgs/games/gamehub/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{ stdenv ,fetchFromGitHub ,
+  meson ,ninja ,vala ,pkg-config ,desktop-file-utils ,
+  glib ,gtk3 ,libgee ,libsoup ,json-glib ,sqlite ,webkitgtk ,libmanette ,libXtst ,wrapGAppsHook
+}:
+stdenv.mkDerivation rec {
+  pname = "GameHub";
+  version = "0.16.0-1";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:8]:
```suggestion
  version = "0.16.0-1";

  src = fetchFromGitHub {
```

File: pkgs/games/gamehub/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{ stdenv ,fetchFromGitHub ,
+  meson ,ninja ,vala ,pkg-config ,desktop-file-utils ,
+  glib ,gtk3 ,libgee ,libsoup ,json-glib ,sqlite ,webkitgtk ,libmanette ,libXtst ,wrapGAppsHook
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:5]:
```suggestion
{ stdenv, fetchFromGitHub
, meson, ninja, vala, pkg-config, desktop-file-utils
, glib, gtk3, libgee, libsoup, json-glib, sqlite, webkitgtk, libmanette, libXtst, wrapGAppsHook
}:

stdenv.mkDerivation rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/games/gamehub/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/games/gamehub/default.nix:48]: Thsi is important as it fixes the CI.

File: pkgs/games/gamehub/default.nix [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+, libgee
+, libsoup
+, json-glib
+, sqlite
+, webkitgtk
+, libmanette
+, libXtst
+, wrapGAppsHook
+}:
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:20]:
```suggestion
}:

stdenv.mkDerivation rec {
```

File: pkgs/games/gamehub/default.nix [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    gtk3
+    libgee
+    libsoup
+    json-glib
+    sqlite
+    webkitgtk
+    libmanette
+    libXtst
+  ];
+  meta = with lib;{
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:49]:
```suggestion
  ];

  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/games/gamehub/default.nix [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    sqlite
+    webkitgtk
+    libmanette
+    libXtst
+  ];
+
+  meta = with lib; {
+    description = "All your games in one place";
+    homepage = "https://tkashkin.tk/projects/gamehub/";
+    mantainers = with mantainers; [ pasqui23 ];
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:56]:
```suggestion
    mantainers = with maintainers; [ pasqui23 ];
```

File: pkgs/games/gamehub/default.nix [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  nativeBuildInputs = [
+    meson
+    ninja
+    vala
+    pkg-config
+    desktop-file-utils
+    wrapGAppsHook
+
+  ];
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/games/gamehub/default.nix:41]: I don't think the usage of propagatedBuildInputs here is correct. Libraries which are linked against should go to buildInputs. Executables which are needed at runtime should be made available using makeWrapper.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by pasqui23** [comment]
Ok, I can't build gamehub anymore somehow and I don't have interest in building it

**Pull request closed without merging**